### ISSUE TITLE
Add resource dependencies in model serving stack

### ISSUE THREAD

**Issue by RichardSieg** [aws-samples/amazon-sagemaker-model-serving-using-aws-cdk#7]
Hey there! 

Thanks for the great repository! I am using it as a template for my own model deployment.

There is one issue with the initial deployment: the endpoint needs the configuration and the configuration needs the model to be fully created. One should therefore add `addDependsOn(theRespectiveResource)` (CDK version 1.137.0) to the configuration and endpoint resources so that they wait for the creation until the other resources are created.

**Comment by kwonyulchoi** [comment]
Hi RichardSieg,
Thanks for your attention.
As you mentioned, SageMaker endpoint/configuration/model have dependency sequentially.
That's right.

By the way, because CDK is a imperative programming approach, deployment is executed step by step from top to bottom within a CDK Stack. So we don't need to declare dependencies explicitly.

Particularly, "addDepensOn()" method is valid on CDK version 1.75.0 which this project is using in "package.json".
The document of "addDepensOn()" method explains when this method is utilized. 
Docs====> "This can be used for resources across stacks (or nested stack) boundaries"
In other words, this method is meaningful for deploying the dependent resources across CDK Stacks.
 
Thanks

**Comment by RichardSieg** [comment]
Hey there :) Thanks for your reply. 

Your statement about the imperative programming approach is not quite correct: CDK merely "compiles" the code to CloudFormation. I could not run the code without using the `addDependsOn`.

The "depends on" relation is also necessary for resources inside the same stack sometimes (like in this case). For a more detailed description of depends on see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html
